### ISSUE TITLE
Move modal scrollbars inside the modal (again)

### ISSUE THREAD

**Issue by LeEnno** [twbs/bootstrap#15735]
This mind sound like a step backward at first, but bear with me: **how about placing the modal scrollbar back to the inner of the modal instead of the window's corner**? Basically as it's been the case in [v2.3.3](http://getbootstrap.com/2.3.2/javascript.html#modals "Javascript · Bootstrap") (with slight modifications).

The current solution (placing the modal's scrollbar at the window's right edge) has a few issues:

- `padding-right` needs to be applied to `body` when a modal opens. This may lead to layout changes sometimes (#15229).
- `.modal-backdrop` had to be moved from `body` to `.modal` for usability reasons (#13816, fixed in #14724). Due to regressions (#14916, #15008) it now also has to be `position:absolute` instead of `fixed`, so its `height` has to be adjusted via JS (#14927). Manually updating height to account for modal's content changes, seems strange to me (and a few others as well, see #15136).
- Users have to use the body's scrollbar (position-wise) to scroll the modal's content. We as bootstrappers may have gotten used to it, but actually placing the scrollbar that so way from the actual content it scrolls, is kinda strange.

**My proposed solution:**

You can see it in action [here](http://materializecss.com/modals.html) (click "Modal" and add content via inspector, so the modal's content becomes scrollable).

*Advantages:*

- No padding adjustments. (#15229)
- No scrollbar hidden under backdrops. (#14916) 
- Scrolling inside the modal scrolls the modal, scrolling outside the modal scrolls the body (just as it works with overlaying windows in your OS). (#15008)
- No manual height updating. (#15136)
- Basically fixes all the mentioned bugs.

Things that would need to change in bootstrap:

- Put scrollbars onto `.modal-content`, not onto `.modal`.
- Give `.modal` a `max-height` and a `top` value (i.e. `max-height:70%; top:10%`).
- Put `.modal-backdrop` back onto `body`.
- Don't hide the `body`'s scrollbar.

*Disadvantages:*

- Scrollbars visually conflict with the modal's rounded corners.
- Scrollbar's width decreases `width` in `.modal-content`. Some sites *may* assume a never-changing modal width, so it *may* break backward compatibility. But so did the current solution, which already leads [some people](https://github.com/twbs/bootstrap/issues/15136#issuecomment-70710126) to not updating bootstrap.

---

I hope all of this is comprehensible. What do you think?

**Comment by martin-g** [comment]
Amen!
The bad behavior of Modal component makes me stay with 3.3.0.

I guess a Pull Request with a demo will make it easier for @twbs/team to review the proposed solution.

**Comment by LeEnno** [comment]
@martin-g a pull request would make thinks easier, for sure. But I don't want to invest time, if the solution will get rejected anyway.

**Comment by DaSchTour** [comment]
Recieved exaclty the same request some days ago from the keyusers of the webshop I work on.

**Comment by mdo** [comment]
Maybe in v4, but definitely not in v3.

**Comment by martin-g** [comment]
@LeEnno I guess your options now is to roll (fork) your own Modal. Please let me know if you go this route.
